Dockerfile

```diff
@@ -7,25 +12,87 @@ RUN apt-get update && apt-get install -y wget \
     git \
     python
 
-ADD . /opt/apalache/
+ADD ./3rdparty /opt/apalache/3rdparty
 WORKDIR /opt/apalache/
 
 # Workaround for Surefire not finding ForkedBooter
 # (see https://stackoverflow.com/questions/53010200/maven-surefire-could-not-find-forkedbooter-class)
 ENV _JAVA_OPTIONS="-Djdk.net.URLClassPath.disableClassPathURLCheck=true"
 
 RUN 3rdparty/install-local.sh --nocache
+
+FROM maven:3.6.3-jdk-8-slim AS builder
+
+# This is the image for building apalache itself
+
+ADD . /opt/apalache/
+WORKDIR /opt/apalache/
+
+# Vendored binaries
+COPY --from=thirdparty /opt/apalache/3rdparty/bin /opt/apalache/3rdparty/bin
+# Vendored libraies
+COPY --from=thirdparty /opt/apalache/3rdparty/lib /opt/apalache/3rdparty/lib
+# The maven repository
+COPY --from=thirdparty /root/.m2 /root/.m2
+
+# Workaround for Surefire not finding ForkedBooter
+# (see https://stackoverflow.com/questions/53010200/maven-surefire-could-not-find-forkedbooter-class)
+ENV _JAVA_OPTIONS="-Djdk.net.URLClassPath.disableClassPathURLCheck=true"
```

Comment:
Similar question, @konnov or @andrey-kuprianov: Are these runtime dependencies or something we need here in the build stage?

Reply:
```docker
# The maven repository
COPY --from=thirdparty /root/.m2 /root/.m2
```

This one is definitely needed for the build stage. The other two are probably needed for the runtime (e.g., dynamic libraries).

Reply:
Sorry, I'm only asking about the environment variable here `_JAVA_OPTIONS=...`, not about copying the maven repository.

Reply:
Yeah, it was an option needed by maven to make the build. But I am not actually sure, whether it is needed anymore.